### ISSUE TITLE
Logger: add form submission parameters

### ISSUE THREAD

**Issue by cllns** [hanami/hanami#733]
Our logger records query parameters, but not parameters submitted via form submissions. 

Current output:
```
[test_app] [INFO] [2017-02-23 16:34:24 -0700] HTTP/1.1 POST 405 ::1 /users/new  - 0.00902
```

Proposed (example) output:
```
[test_app] [INFO] [2017-02-23 16:34:24 -0700] HTTP/1.1 POST 405 ::1 /users/new - 0.00902
[test_app] [INFO]     Parameters: { "email" => "hello@example.com", "password" => "[FILTERED] } 
```

Note that we ought to filter out `password`. We should also filter out `password_confirmation` (maybe just filter out all keys that match `/^password/`?). 
We should also allow people to specify other field names (or regexes?) to filter out. For example, so people could filter out `credit_card_number`.

When printing to STDOUT, like in development mode, I personally prefer displaying the parameters hash with `pp`, so it's easy to read. Printing a hash on a single line is incredibly error-prone, in my experience. 
Should we also `deep_symbolize` the params before outputting them?

Note: this is a proposed enhancement for `v1.1`, **not** `v1.0`.

What do you all think? 💁

---

Currently assigned to @gizotti

**Comment by b264** [comment]
I agree with all of that, especially the `v1.1` part.

I do have a tip-- as a Hanami beginner, if I saw fields being filtered out, it would be great if the search term on how to change the setting was the thing displayed in the log, example:
```
[test_app] [INFO]     Parameters: { "email" => "hello@example.com", "password" => "[KEYS_HIDDEN] }
```
so then if I search for, example, `https://ddg.gg/hanami%20%22KEYS_HIDDEN%22` then I get a search result saying where to adjust the regex `Hanami::Logger::KEYS_HIDDEN` or whatnot.  Just a suggestion that how to change it is searchable from the filtered text output itself with a word a little less generic than "FILTERED" and especially not the same word Rails uses.

**Comment by davydovanton** [comment]
Great idea 👍
I think we need to use `pp` style for displaying params (because sometimes params are too long).

**Comment by gizotti** [comment]
@cllns @davydovanton I'll have a go at this one. Can you guys point out a good start point, I am looking for it now but a little help is always welcome :).

As for `pp`, should we also have an option if people want to disable `pp` and display the params in a single line?

**Comment by gizotti** [comment]
Hey guys. Just a heads up that I started on this one.
I'll try to put something out sometime this week :)

**Comment by jodosha** [comment]
Implemented by #818
